examples/server/server.cpp

```diff
@@ -510,132 +529,245 @@ bool server_params_parse(int argc, char **argv, server_params &sparams, gpt_para
   }
-  if (!body["n_keep"].is_null())
-  {
-    llama.params.n_keep = body["n_keep"].get<int>();
+  if (!body["presence_penalty"].is_null()) {
+    llama.params.presence_penalty = body["presence_penalty"].get<float>();
+  } else {
+    llama.params.presence_penalty = default_params.presence_penalty;
   }
-  if (!body["as_loop"].is_null())
-  {
-    llama.as_loop = body["as_loop"].get<bool>();
+  if (!body["frequency_penalty"].is_null()) {
+    llama.params.frequency_penalty = body["frequency_penalty"].get<float>();
+  } else {
+    llama.params.frequency_penalty = default_params.frequency_penalty;
   }
-  if (!body["interactive"].is_null())
-  {
-    llama.params.interactive = body["interactive"].get<bool>();
+  if (!body["mirostat"].is_null()) {
+    llama.params.mirostat = body["mirostat"].get<int>();
+  } else {
+    llama.params.mirostat = default_params.mirostat;
   }
-  if (!body["prompt"].is_null())
-  {
-    llama.params.prompt = body["prompt"].get<std::string>();
+  if (!body["mirostat_tau"].is_null()) {
+    llama.params.mirostat_tau = body["mirostat_tau"].get<float>();
+  } else {
+    llama.params.mirostat_tau = default_params.mirostat_tau;
   }
-  else
-  {
-    json data = {
-        {"status", "error"},
-        {"reason", "You need to pass the prompt"}};
-    res.set_content(data.dump(), "application/json");
+  if (!body["mirostat_eta"].is_null()) {
+    llama.params.mirostat_eta = body["mirostat_eta"].get<float>();
+  } else {
+    llama.params.mirostat_eta = default_params.mirostat_eta;
+  }
+  if (!body["penalize_nl"].is_null()) {
+    llama.params.penalize_nl = body["penalize_nl"].get<bool>();
+  } else {
+    llama.params.penalize_nl = default_params.penalize_nl;
+  }
+  if (!body["n_keep"].is_null()) {
+    llama.params.n_keep = body["n_keep"].get<int32_t>();
+  } else {
+    llama.params.n_keep = default_params.n_keep;
+  }
+  if (!body["seed"].is_null()) {
+    llama.params.seed = body["seed"].get<int32_t>();
+  } else {
+    llama.params.seed = time(NULL);
+  }
+
+  llama.params.logit_bias.clear();
+  if (!body["ignore_eos"].is_null() && body["ignore_eos"].get<bool>()) {
+    llama.params.logit_bias[llama_token_eos()] = -INFINITY;
+  }
+  if (body["logit_bias"].is_array()) {
+    int n_vocab = llama_n_vocab(llama.ctx);
+    for (const auto &el : body["logit_bias"]) {
+      if (el.is_array() && el.size() == 2 && el[0].is_number_integer()) {
+        llama_token tok = el[0].get<llama_token>();
+        if (tok >= 0 && tok < n_vocab) {
+          if (el[1].is_number_float()) {
+            llama.params.logit_bias[tok] = el[1].get<float>();
+          } else if (el[1].is_boolean() && !el[1].get<bool>()) {
+            llama.params.logit_bias[tok] = -INFINITY;
+          }
+        }
+      }
+    }
+  }
+
+  if (!body["prompt"].is_null()) {
+    llama.params.prompt = body["prompt"].get<std::string>();
+  } else {
+    json data = {{"status", "error"}, {"reason", "You need to pass the prompt"}};
```

Comment:
```suggestion
    json data = {{"status", "error"}, {"reason", "You need to provide a prompt"}};
```

In any case, this behavior is different from the `main` executable. As a developer I frequently pass empty prompts just to test correctness/performance but as I user I don't care.

Reply:
That is a good point. Technically it only needs `<BOS>` to get generating.